### ISSUE TITLE
Support Android Targets

### ISSUE THREAD

**Issue by leleliu008** [dbrgn/tealdeer#260]
https://github.com/dbrgn/tealdeer/archive/v1.5.0.tar.gz

![21](https://user-images.githubusercontent.com/3040852/149950669-5295f210-72b9-4d54-9395-54e528593ede.PNG)

**Comment by dbrgn** [comment]
Android is not currently supported. Just out of curiosity, what kind of Android device do you have that would benefit from a tldr installation?

**Comment by leleliu008** [comment]
HUAWEI MatePad Pro
![matepad-pro](https://user-images.githubusercontent.com/3040852/149956737-1bdef0f8-f478-4c41-ab04-c4bcd987399b.jpg)

**Comment by dbrgn** [comment]
Ok 🙂 I guess it shouldn't be too hard to add proper Android support.

**Comment by leleliu008** [comment]
```patch
--- a/src/types.rs
+++ b/src/types.rs
@@ -43,7 +43,7 @@
 }
 
 impl PlatformType {
-    #[cfg(target_os = "linux")]
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     pub fn current() -> Self {
         Self::Linux
     }
@@ -66,6 +66,7 @@
 
     #[cfg(not(any(
         target_os = "linux",
+        target_os = "android",
         target_os = "macos",
         target_os = "freebsd",
         target_os = "netbsd",
```

**Comment by dbrgn** [comment]
A pull request would be welcome. Note that "android" is a dedicated platform type supported by tldr (with some Android-specific pages available), so the `PlatformType` should be extended with an `Android` variant (and not just fall back to Linux).

**Comment by niklasmohrin** [comment]
A PR for this should also add a job to github actions that makes sure the build still works for android; I am not sure whether we can test in a simulated android environment, a little research from future implementers would be nice :)

**Comment by leleliu008** [comment]
We usually just check if can be built for Android and do not check if can be run on Android. Because ARM-based emulators were too slow. but In the real world, 99% Android devices are ARM-based.

**Comment by dbrgn** [comment]
Android support should be working since #267 was merged. Could you test building the `main` branch?

**Comment by niklasmohrin** [comment]
I wouldn't consider this issue done until we build in CI, it shouldn't be difficult (see [my comment in the other PR](https://github.com/dbrgn/tealdeer/pull/274#discussion_r897300717)). I don't think we need to run tests for android in CI, but if its cheap we can do that too.

For me, #267 was mainly about the platform type being available on desktop builds, so I didn't require it there, but this is about supporting the platform which I think needs CI as a safety net. Reopening for now

**Comment by leleliu008** [comment]
I confired that https://github.com/dbrgn/tealdeer/commit/543fa2290d4866d677325a7c8f2a7d7a3e02d838 can be successfully built for Android. I didn't check if can be successfully run on Android device. I think if have problems, end-users will report bugs.

**Comment by dbrgn** [comment]
@leleliu008 thanks for testing!

@niklasmohrin I agree that having CI would be nice to have.

**Comment by leleliu008** [comment]
run `trdl -l` command on Android device failed.

https://github.com/termux/termux-packages/pull/9072
https://github.com/termux/termux-packages/blob/master/packages/tealdeer/Cargo.toml.patch

**Comment by Walker-00** [comment]
it's working fine on android (termux) with `cargo install tealdeer`. should close?

**Comment by niklasmohrin** [comment]
No, see https://github.com/tealdeer-rs/tealdeer/issues/260#issuecomment-1155724250
